internal/match/resyntax_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package match
+
+import (
+	"strings"
+	"testing"
+)
+
+var reParseTests = []struct {
+	in  string
+	err string
+	out string
+}{
+	{in: "abc", out: "abc"},
+	{in: "abc//**text**//def", out: "abc def"},
+	{in: "a b c", out: "a b c"},
+	{in: " (( abc )) ??", out: "((abc))??"},
+	{in: "a b (( c ))??", out: "a b\n((c))??"},
+	{in: "(a b ((c) ))??", out: "a b\n((c))??"},
```

Comment:
Why not a syntax error?

Reply:
It just seems like more work for little benefit.
You do want to be able to put (( )) around parenthesized text.
Thats the whole reason for using the doubled symbols instead of the usual single ones.
If it commonly led to a mistake, then sure, but if there's a mistake your (( )) probably won't balance anyway.